[repo] docs: added concepts to the getting started docs (#1176)
microsoft/teams-ai: getting-started/CONCEPTS/ACTION-PLANNER.md

# Action Planner

> The Action Planner is a powerful planner that uses a LLM to generate plans. It can trigger parameterized actions and send text based responses to the user. It supports the following advanced features:

Reviewer: ```suggestion
The Action Planner is a powerful planner that uses an LLM to generate plans. It can trigger parameterized actions and send text-based responses to the user. It supports the following advanced features:
```

### [Prompt Management](./PROMPTS.md)

> The Action Planner has a built-in prompt management system that supports creating prompt templates as folders in the file system. A prompt template is the prompt text along with all the configurations for completion with the LLM model. Dynamic prompts are also supported with template variables and functions.

Reviewer: ```suggestion
The Action Planner has a built-in prompt management system that supports creating prompt templates as folders in the file system. A prompt template is the prompt text along with all the configurations for completion with the LLM model. Dynamic prompts also support template variables and functions.
```

### [Augmentations](./AUGMENTATIONS.md)
Augmentations virtually eliminate the need for prompt engineering. Prompts
can be configured to use a named augmentation which will be automatically appended to the outgoing
prompt. Augmentations let the developer specify whether they want to support multi-step plans (sequence),
or create an AutoGPT style agent (monologue).

### Validations 
Validators are used to validate the response returned by the LLM and can guarantee

> that the parameters passed to an action mach a supplied schema. The validator used is automatically

Reviewer 2: small typo - match

> that the parameters passed to an action mach a supplied schema. The validator used is automatically

Reviewer: ```suggestion
that the parameters passed to an action match a supplied schema. The validator used is automatically
```
@lilyydu you can use the suggestions button: 
<img width="65" alt="image" src="https://github.com/microsoft/teams-ai/assets/14900841/01881345-dbf5-4089-b8bb-cdb92fd8727c">

> selected based on the augmentation being used. Validators also prevent hallucinated action names

Reviewer: ```suggestion
selected based on the augmentation being used. Validators also prevent hallucinated action names,
```
